src/globus_cli/login_manager/tokenstore.py

```diff
@@ -140,7 +149,15 @@ def delete_templated_client():
     adapter = token_storage_adapter()
 
     # first, get the templated credentialed client
-    ac = internal_auth_client()
+    try:
+        ac = internal_auth_client()
+
+    # client logins shouldn't have templated clients to delete,
+    # but just in-case there was one catch errors here and return
+    # rather than skip cleanup entirely
+    except ValueError:
+        if is_client_login():
+            return
```

Comment:
Right now, this check would discard any `ValueError` and fallthrough if `is_client_login() == False`.
We want to fix that, but I'm also looking at the overall "flow" and wondering if it might be better to leave this method unmodified.

Instead of making `delete_templated_client()` work when `is_client_login() == True`, what about letting it fail, and updating `globus logout` to ensure that we don't end up calling it?
Otherwise, when reading the source for logout, we see a method call which "looks wrong" (because there shouldn't be a client to delete), and it's not immediately obvious that it's a no-op.